python/lsst/source/injection/inject_base.py

```diff
@@ -0,0 +1,636 @@
+        equivalents; any source profile types that are not GalSim classes will
+        be flagged.
+
+        Note: Unlike the cleaning method, no sources are actually removed here.
+        Instead, a binary flag is set in the *injection_flag* column for each
+        source. Only unflagged sources will be generated for source injection.
+
+        Parameters
+        ----------
+        injection_catalog : `astropy.table.Table`
+            Catalog of sources to be injected.
+        input_exposure : `lsst.afw.image.exposure.ExposureF`
+            The exposure sources will be injected into.
+        binary_flags : `dict` [`str`, `int`]
+            Dictionary of binary flags to be used in the injection_flag column.
+
+        Returns
+        -------
+        injection_catalog : `astropy.table.Table`
+            The cleaned catalog of sources to be injected.
+        """
+        self.config = cast(BaseInjectConfig, self.config)
+
+        # Flag erroneous magnitude values (missing mag data or NaN mag values).
+        if "mag" not in injection_catalog.columns:
+            # Check injection_catalog has a mag column.
+            self.log.warning("No magnitude data found in injection catalog; cannot inject any sources!")
+            injection_catalog["injection_flag"] += 2 ** binary_flags["MAG_BAD"]
+        else:
+            # Check that all input mag values are finite.
+            mag_array = np.isfinite(ma.array(injection_catalog["mag"]))
+            bad_mag = ~(mag_array.data * ~mag_array.mask)
+            if (num_bad_mag := np.sum(bad_mag)) > 0:
+                grammar = "source" if num_bad_mag == 1 else "sources"
+                self.log.warning(
+                    "Flagging %d injection %s that do not have a finite magnitude.", num_bad_mag, grammar
+                )
+                injection_catalog["injection_flag"][bad_mag] += 2 ** binary_flags["MAG_BAD"]
+
+        # Replace legacy source types with standardized profile names.
+        injection_catalog["source_type"] = injection_catalog["source_type"].astype("O")
+        replace_dict = {
+            "0": "DeltaFunction",
+            "star": "DeltaFunction",
+            "Star": "DeltaFunction",
+            "1": "Sersic",
+            "galaxy": "Sersic",
+            "Galaxy": "Sersic",
+            "2": "Trail",
+            "trail": "Trail",
+            "stamp": "Stamp",
+        }
+        for legacy_type, standard_type in replace_dict.items():
+            legacy_matches = injection_catalog["source_type"] == legacy_type
+            if np.any(legacy_matches):
+                injection_catalog["source_type"][legacy_matches] = standard_type
+        injection_catalog["source_type"] = injection_catalog["source_type"].astype(str)
+
+        # Flag source types not supported by GalSim.
+        input_source_types = set(injection_catalog["source_type"])
+        allowed_source_types = set(dir(galsim) + ["Trail", "Stamp"])
```

Comment:
There's a lot more in the `galsim` namespace than just names of surface brightness profiles.  And even some things that are sbprofiles in an abstract sense won't work here because they're compound/derived (e.g., galsim.Sum, galsim.Deconvolution).  I think I'd just list the allowed profiles here for now: `['Gaussian', 'Moffat', 'Airy', 'Kolmogorov', 'Box', 'TopHat', 'Exponential', 'Sersic', 'Spergel', 'DeltaFunction', 'InclinedExponential', 'InclinedSersic', 'VonKarman']`